setup/main.go

```diff
@@ -0,0 +1,1653 @@
+		if err != nil {
+			return err
+		}
+		selectedPrefix := supportedLogLinePrefixes[prefixIdx]
+		return applyConfigSetting("log_line_prefix", pq.QuoteLiteral(selectedPrefix), state.QueryRunner)
+	},
+}
+
+func validatePath(ans interface{}) error {
+	ansStr, ok := ans.(string)
+	if !ok {
+		return errors.New("expected string value")
+	}
+	// TODO: also confirm this is readable by the regular pganalyze user
+	_, err := os.Stat(ansStr)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var configureLogLocation = &Step{
+	Description: "Determine log location",
+	Check: func(state *SetupState) (bool, error) {
+		if state.SkipLogInsights {
+			return true, nil
+		}
+		return state.CurrentSection.HasKey("db_log_location"), nil
+	},
+	Run: func(state *SetupState) error {
+		guessedLogLocation, err := discoverLogLocation(state.CurrentSection, state.QueryRunner)
+		if err != nil {
+			return err
+		}
+		var logLocationConfirmed bool
+		err = survey.AskOne(&survey.Confirm{
+			Message: fmt.Sprintf("Your database log file or directory appears to be %s; is this correct (will be saved to collector config)?", guessedLogLocation),
+			Default: false,
+		}, &logLocationConfirmed)
+		if err != nil {
+			return err
+		}
+		var logLocation string
+		if logLocationConfirmed {
+			logLocation = guessedLogLocation
+		} else {
+			err = survey.AskOne(&survey.Input{
+				Message: "Please enter the Postgres log file location (will be saved to collector config)",
+			}, &logLocation, survey.WithValidator(validatePath))
+		}
+		_, err = state.CurrentSection.NewKey("db_log_location", logLocation)
+		if err != nil {
+			return err
+		}
+		return state.Config.SaveTo(state.ConfigFilename)
+	},
+}
+
+var confirmAutoExplainSetup = &Step{
+	Description: "Check whether to configure Automated EXPLAIN",
+	Check: func(state *SetupState) (bool, error) {
+		return state.AskedAutomatedExplain, nil
+	},
+	Run: func(state *SetupState) error {
+		var setUpExplain bool
+		err := survey.AskOne(&survey.Confirm{
+			Message: "Proceed to configuring optional Automated EXPLAIN feature?",
+			Help:    "Learn more at https://pganalyze.com/postgres-explain",
+			Default: false,
+		}, &setUpExplain)
+		if err != nil {
+			return err
+		}
+		state.AskedAutomatedExplain = true
+		state.SkipAutomatedExplain = !setUpExplain
+
+		return nil
+	},
+}
+
+var checkUseLogBasedExplain = &Step{
+	Description: "Check whether to use the auto_explain module or log-based EXPLAIN",
+	Check: func(state *SetupState) (bool, error) {
+		if state.SkipAutomatedExplain {
+			return true, nil
+		}
+		return state.CurrentSection.HasKey("enable_log_explain"), nil
+	},
+	Run: func(state *SetupState) error {
+		var optIdx int
+		err := survey.AskOne(&survey.Select{
+			Message: "Select automated EXPLAIN mechanism to use (will be saved to collector config):",
+			Help:    "Learn more about the options at https://pganalyze.com/docs/explain/setup",
+			Options: []string{"auto_explain (recommended)", "log-based EXPLAIN"},
```

Comment:
I would use `Log-based EXPLAIN` here (i.e. uppercase `L`), for consistency with the docs